client/src/components/BulkCreateForm/BulkCreateForm.js

```diff
@@ -76,164 +172,298 @@ export default class BulkCreateForm extends React.Component {
      * @param resetValue - If true, also resets the values of the inputs
      */
     resetForm(resetValue) {
-        const { fileUpload } = this.state;
+        const { fileUpload, fields } = this.state;
+        const { onResetForm } = this.props;
         let { fileUploadInputKey } = this.state;
 
         fileUpload.isValid = true;
         fileUpload.errorMessage = '';
 
         if (resetValue) {
-            fileUpload.file = {};
+            fileUpload.csv = null;
+            fileUpload.zip = null;
+            fileUpload.images = [];
             fileUpload.message = 'No file chosen';
             fileUploadInputKey++;
         }
 
-        this.setState({fileUpload, fileUploadInputKey});
+        fields.forEach(field => field.selected = false);
+
+        onResetForm();
+        this.setState({
+            fileUpload,
+            fileUploadInputKey,
+            fields,
+            showFieldMapping: false,
+            mapping: []
+        });
```

Comment:
Any reason for calling `OnResetForm()` before `this.setState()`?

Reply:
No, don't think it really matters. `onResetForm` just tells the parent to update the props it passes down into this component